src/main/integrations/renderer-profiling.ts

```diff
@@ -0,0 +1,120 @@
+import { NodeClient } from '@sentry/node';
+import { Event, Integration, Profile } from '@sentry/types';
+import { forEachEnvelopeItem, LRUMap } from '@sentry/utils';
+import { app } from 'electron';
+
+import { normaliseProfile } from '../../common';
+import { getDefaultEnvironment, getDefaultReleaseName } from '../context';
+import { ELECTRON_MAJOR_VERSION } from '../electron-normalize';
+import { ElectronMainOptionsInternal } from '../sdk';
+
+// A cache of renderer profiles which need attaching to events
+let RENDERER_PROFILES: LRUMap<string, Profile> | undefined;
+
+/**
+ * Caches a profile to later be re-attached to an event
+ */
+export function rendererProfileFromIpc(event: Event, profile: Profile): void {
+  if (!RENDERER_PROFILES) {
+    return;
+  }
+
+  const profile_id = profile.event_id;
+  RENDERER_PROFILES.set(profile_id, profile);
+
+  if (event) {
+    event.contexts = {
+      ...event.contexts,
+      // Re-add the profile context which we can later use to find the correct profile
+      profile: {
+        profile_id,
+      },
+    };
+  }
+}
+
+/**
+ * Injects 'js-profiling' document policy headers and ensures that profiles get forwarded with transactions
+ */
+export class RendererProfiling implements Integration {
+  /** @inheritDoc */
+  public static id: string = 'RendererProfiling';
+
+  /** @inheritDoc */
+  public readonly name: string;
+
+  public constructor() {
+    this.name = RendererProfiling.id;
+  }
+
+  /** @inheritDoc */
+  public setupOnce(): void {
+    //
+  }
+
+  /** @inheritDoc */
+  public setup(client: NodeClient): void {
+    const options = client.getOptions() as ElectronMainOptionsInternal;
+    if (!options.enableRendererProfiling) {
```

Comment:
Can we instead gate this by `profilesSampleRate` being defined?

Reply:
There are 2 processes here. Using the sample rate would enable profiling in the main process.

Reply:
@timfish Instead of requiring users to both enable renderer profiling bool and BrowserProfiling integration, can we automatically inject BrowserProfiling if we can infer enableRendererProfiling bool is set to true? I'm not sure if this is possible in the concept of electron and the rendering model, but it would allow us to reduce the config down to a single configuration point.